### PR TITLE
hardware: add a setting to choose the serial protocol

### PR DESCRIPTION
**Pull request by jpcima**
#153

I add the drop-down menu which permits to choose "Arduino OPL2" as a selection of serial protocol.
Adding future protocols should take minimal effort from here.

### CODE REVIEW

File: src/opl/chips/opl_serial_port.cpp [hunk 1]
```diff
@@ -52,18 +56,23 @@ void OPL_SerialPort::writeReg(uint16_t addr, uint8_t data)
...
-    ///
-    uint8_t sendBuffer[2] = {(uint8_t)addr, (uint8_t)data};
-    port->write((char *)sendBuffer, 2);
+    unsigned protocol = m_protocol.load();
+
+    switch(protocol)
+    {
+    // TODO support OPL2 only
+    default:
+    case ProtocolArduinoOPL2:
```
Comment (Wohlstand) [inline src/opl/chips/opl_serial_port.cpp]: Let's ask @nukeykt for his OPL3 rig protocol? The thing would be able to work on his thing also 🤔

Comment (nukeykt) [reply src/opl/chips/opl_serial_port.cpp]: It's pretty simple:
```
            {
                xx[0 + size] = (regg >> 6) | 0x80;
                xx[1 + size] = ((regg & 0x3f) << 1) | (data >> 7);
                xx[2 + size] = (data & 0x7f);
            }
```
```
  if(Serial.available())
  {
    byte x1 = Serial.read();
    if(x1&0x80)
    {
      while(!Serial.available());
      byte x2= Serial.read();
      while(!Serial.available());
      byte x3 = Serial.read();
      byte reg = ((x1<<6)&0xc0)|((x2>>1)&0x3f);
      byte data = ((x2<<7)&0x80)|(x3&0x7f);
      writereg(reg,data);
    }
  }
```

Comment (Wohlstand) [reply src/opl/chips/opl_serial_port.cpp]: Thanks :fox_face:, and thanks @jpcima for implementation! :fox_face: 
If you want, feel free to check the thing in action if you can :fox_face: :+1:

Comment (nukeykt) [reply src/opl/chips/opl_serial_port.cpp]: Cool, thanks. Unfortunately my OPL3 rig was prototyped on breadboard and i disassembled it years ago to use breadboard for other projects

Comment (Wohlstand) [reply src/opl/chips/opl_serial_port.cpp]: Sad, however, yeah, someone who'll assemble something like, will be able to use the protocol for playback :fox_face:

**Comment by pcky** [comment]
FYI - I guess, I'm the one who's now taking advantage of this protocol implementation. @DhrBaksteen released a new OPL3 board, called OPL3 Duo!. I adapted @nukeykt's protocol for it to get it connected to OPL3BankEditor, DosBox and Adtrack2:
[https://github.com/pcky/ArduinoOPL2/commit/b3800c30aa9e7a2fefc104b57a9c9ac4cd2274e3](url)

**Comment by DhrBaksteen** [comment]
Sorry upfront because this will be a bit off-topic for this repo...

Thanks for tagging me @pcky because I wasn't aware of @nukeykt's protocol. I put together a serial example really quickly a few days ago because people were asking for some example code and I decided to leave any though about a proper protocol to deal with desynchronization for later 😅. Seems @nukeykt's has already solved that issue so I think it would be good to replace the current serial example for the OPL3 Duo with his implementation. Maybe use bit 3 (0x08) of the fist byte to select between the first or the second YMF262?

Feel free to submit a PR for the Arduino library because there are many people using the OPL3BankEditor already with the OPL2 board and I'm sure they want to do the same with my new OPL3 Duo board.

**Pull request merged**